### PR TITLE
Fixing checking function for archives

### PR DESCRIPTION
**Pull request by jordancorns**
This PR aims to fix bad checking function `is_archive.`

I propose two functions `is_archive_in` and `is_archive_out.`:

- The first one check the extensions and if the archive is readable with libs `ziplib` and `tarlib`, not corrupted. (e.g. A file that has a `.zip` extension but not real `zip`)

- The second one `is_archive_out` check only the extension of the file.

I encountered the case of an corrupted `.zip` archive with CATI Collector,  . Deidentification module believe that this archive was a TAR because of the old `is_archive` function

I've add some tests with pytest for this fixed functionnality.

### CODE REVIEW

File: tests/test_archive.py [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
+def opening_archive_path(request):
+    return request.param
+
+
+@pytest.fixture(params=glob.glob(ARCHIVES_NOT_OPENING_DIR+'/*'))
+def not_opening_archive_path(request):
     return request.param
 
 
-@pytest.fixture(params=os.listdir('tests/data/files'))
```
Comment (Hboni) [inline tests/test_archive.py:10]: I was wondering what could be the best between `os.listdir` and `glob`, and I found something about [os.scandir](https://docs.python.org/3/library/os.html#os.scandir), just to mention it. In our case, any of those solutions are good.

File: deidentification/archive.py [hunk 2]
```diff
@@ -2,34 +2,54 @@
 import tarfile
 import zipfile
 import os
+from deidentification import DeidentificationError
 
 
-def is_archive(filename):
+
+def is_archive_in(filepath):
```
Comment (Hboni) [inline deidentification/archive.py:9]: What do you think about renaming `is_archive_..`functions to be more explicit about what they doing?
for example :

`is_archive_in`  --> `is_archive_file`
`is_archive_out` --> `is_archive_ext` or `is_archive_extension`

Comment (jordancorns) [reply deidentification/archive.py:9]: I think that a you have a good idea ! I was looking for a name more explicit.

File: tests/test_archive.py (hunk 1 above)
Comment (jordancorns) [reply tests/test_archive.py:10]: Thanks, I'll check `scandir` in the future

**Pull request merged**